### PR TITLE
 html-->markdown web/api/abortsignal

### PR DESCRIPTION
**Pull request by fwqaaq**

### CODE REVIEW

File: files/zh-cn/web/api/abortsignal/index.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  - : 返回一个在指定事件后将自动终止的 **`AbortSignal`** 实例。
+
+### 事件
+
+使用 [`addEventListener()`](/zh-CN/docs/Web/API/EventTarget/addEventListener) 或将事件监听器分配给该接口的 `oneventname` 属性。
+
+- [`abort`](/zh-CN/docs/Web/API/AbortSignal/abort_event)
+  - : 当与 signal 通信 DOM 请求被中止时调用。也可以通过 `onabort` 属性调用。
+- {{domxref("AbortSignal.onabort")}}
+  - : 当 {{event("abort_(dom_abort_api)", "abort")}} 事件触发时，即当信号正在与之通信的 DOM 请求被中止时调用。
```
Comment (yin1999) [inline files/zh-cn/web/api/abortsignal/index.md:39]:
```suggestion
```

File: files/zh-cn/web/api/abortsignal/index.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+以下片段展示了我们如何使用 signal 去中止使用 [Fetch API](/zh-CN/docs/Web/API/Fetch_API) 下载视频。
+
+我们首先使用 {{domxref("AbortController.AbortController", "AbortController()")}} 创建一个中止控制器，然后使用 {{domxref("AbortController.signal")}} 属性获取与它联系的 {{domxref("AbortSignal")}} 对象的引用。
+
+当 [fetch 请求](/zh-CN/docs/Web/API/fetch) 开始时，我们将 `AbortSignal` 作为一个选项传递进请求的选项对象中（见下面的 `{signal}`）。这个将信号和控制器与 fetch 请求相联系，并且允许我们通过调用 {{domxref("AbortController.abort()")}} 中止它。你可以看见当中止按钮（`abortBtn`）被点击，第二个事件监听器触发，fetch 操作被中止。
+
+```js
+const controller = new AbortController();
+const signal = controller.signal;
+const url = 'video.mp4';
```
Comment (yin1999) [inline files/zh-cn/web/api/abortsignal/index.md:54]: please note the empty line in code sample, please copy it from github's editor (not preview)

```suggestion
const signal = controller.signal;

const url = 'video.mp4';
```

File: files/zh-cn/web/api/abortsignal/index.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    })
+    .catch((err) => {
+      console.error(`Download error: ${err.message}`);
+    });
+}
+```
+
+> **备注：** 当调用 `abort()` 时，`fetch()` 会调用 reject，返回一个“`AbortError`”`DOMException`。
+
+你可以 [在 github 上找到一个完整的可用运行示例](https://github.com/mdn/dom-examples/tree/master/abort-api)；你也可以参见[现场演示](https://mdn.github.io/dom-examples/abort-api/)。
```
Comment (yin1999) [inline files/zh-cn/web/api/abortsignal/index.md:75]:
```suggestion
你可以[在 github 上找到一个完整、可运行的示例](https://github.com/mdn/dom-examples/tree/master/abort-api)；你也可以参见[在线演示](https://mdn.github.io/dom-examples/abort-api/)。
```

File: files/zh-cn/web/api/abortsignal/index.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+{{Specifications}}
+
+## 浏览器兼容性
+
+{{Compat}}
+
+## 参见
+
+- [Fetch API](/en-US/docs/Web/API/Fetch_API)
```
Comment (yin1999) [inline files/zh-cn/web/api/abortsignal/index.md:141]:
```suggestion
- [Fetch API](/zh-CN/docs/Web/API/Fetch_API)
```

File: files/zh-cn/web/api/abortsignal/index.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+- {{domxref("AbortSignal.onabort")}}
+  - : 当 {{event("abort_(dom_abort_api)", "abort")}} 事件触发时，即当信号正在与之通信的 DOM 请求被中止时调用。
+
+## 示例
+
+### 使用显示的信号终止 fetch 操作
+
+以下片段展示了我们如何使用 signal 去中止使用 [Fetch API](/zh-CN/docs/Web/API/Fetch_API) 下载视频。
+
+我们首先使用 {{domxref("AbortController.AbortController", "AbortController()")}} 创建一个中止控制器，然后使用 {{domxref("AbortController.signal")}} 属性获取与它联系的 {{domxref("AbortSignal")}} 对象的引用。
```
Comment (yin1999) [inline files/zh-cn/web/api/abortsignal/index.md:47]:
```suggestion
我们首先使用 {{domxref("AbortController.AbortController", "AbortController()")}} 创建一个中止控制器，然后使用 {{domxref("AbortController.signal")}} 属性获取与它关联的 {{domxref("AbortSignal")}} 对象的引用。
```

File: files/zh-cn/web/api/abortsignal/index.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+当 [fetch 请求](/zh-CN/docs/Web/API/fetch) 开始时，我们将 `AbortSignal` 作为一个选项传递进请求的选项对象中（见下面的 `{signal}`）。这个将信号和控制器与 fetch 请求相联系，并且允许我们通过调用 {{domxref("AbortController.abort()")}} 中止它。你可以看见当中止按钮（`abortBtn`）被点击，第二个事件监听器触发，fetch 操作被中止。
```
Comment (yin1999) [inline files/zh-cn/web/api/abortsignal/index.md:49]:
```suggestion
当 [fetch 请求](/zh-CN/docs/Web/API/fetch)开始时，我们将 `AbortSignal` 作为一个选项传递进请求的 option 对象中（见下面的 `{signal}`）。这个将信号和控制器与 fetch 请求相关联，并且允许我们通过调用 {{domxref("AbortController.abort()")}} 中止它。你可以看见当中止按钮（`abortBtn`）被点击，第二个事件监听器触发，fetch 操作被中止。
```

File: files/zh-cn/web/api/abortsignal/index.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+    .then((response) => {
+      console.log('Download complete', response);
+    })
+    .catch((err) => {
+      console.error(`Download error: ${err.message}`);
+    });
+}
+```
+
+> **备注：** 当调用 `abort()` 时，`fetch()` 会调用 reject，返回一个“`AbortError`”`DOMException`。
```
Comment (yin1999) [inline files/zh-cn/web/api/abortsignal/index.md:73]:
```suggestion
> **备注：** 当调用 `abort()` 时，`fetch()` promise 会以“`AbortError`”`DOMException` 拒绝。
```

File: files/zh-cn/web/api/abortsignal/index.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+你可以 [在 github 上找到一个完整的可用运行示例](https://github.com/mdn/dom-examples/tree/master/abort-api)；你也可以参见[现场演示](https://mdn.github.io/dom-examples/abort-api/)。
+
+### 中止超时的读取操作
+
+如果你需要中止一个超时的操作，你可以使用 {{domxref("AbortSignal.timeout()")}} 静态方法。该方法返回一个 `AbortSignal` 并在指定的超时毫秒后自动中止。
```
Comment (yin1999) [inline files/zh-cn/web/api/abortsignal/index.md:79]:
```suggestion
如果你需要中止一个超时的操作，你可以使用 {{domxref("AbortSignal.timeout()")}} 静态方法。该方法返回一个 `AbortSignal` 并在指定的毫秒时间后后自动超时。
```

File: files/zh-cn/web/api/abortsignal/index.md [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+
+以下代码片段展示了如何成功地下载一个文件或者在五秒钟后处理一个超时的错误。注意，当出现超时时，`fetch()` promise 会以“`TimeoutError`”`DOMException` 拒绝。这允许代码区分超时（可能需要用户通知）和用户中止。
```
Comment (yin1999) [inline files/zh-cn/web/api/abortsignal/index.md:81]:
```suggestion
以下代码片段展示了如何成功地下载一个文件或者在五秒钟后处理一个超时的错误。注意，当出现超时时，`fetch()` promise 会以“`TimeoutError`”`DOMException` 拒绝。这允许代码区分超时（可能需要通知用户）和用户中止。
```

File: files/zh-cn/web/api/abortsignal/index.md [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+  } else if (err.name === "TypeError") {
+    console.error("AbortSignal.timeout() method is not supported");
+  } else {
+    // A network error, or some other problem.
+    console.error(`Error: type: ${err.name}, message: ${err.message}`);
+  }
+}
+```
+
+### 超时或显示中止 fetch
```
Comment (yin1999) [inline files/zh-cn/web/api/abortsignal/index.md:103]:
```suggestion
### 超时或显式中止 fetch
```

File: files/zh-cn/web/api/abortsignal/index.md [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+
+`fetch()` 不是为了组合多个信号而设计的，所以你不能因为调用了 {{domxref("AbortController.abort()")}} 或者 `AbortSignal` 超时而"直接"中止下载（尽管在上面的例子中，如果由停止按钮等内置浏览器机制触发，超时信号将中止）。
+
+为了触发多个 signal，它们必须是 daisy 链式的连接。以下代码片段展示了在处理程序中如何为独立的计时器调用 {{domxref("AbortController.abort()")}}。
```
Comment (yin1999) [inline files/zh-cn/web/api/abortsignal/index.md:107]:
```suggestion
为了触发多个 signal，它们必须是菊花链式连接。以下代码片段展示了在处理程序中如何为独立的计时器调用 {{domxref("AbortController.abort()")}}。
```

**Review by yin1999 — CHANGES_REQUESTED** [review]

File: files/zh-cn/web/api/abortsignal/index.md (hunk 1 above)
Comment (Jack-Zhang-1314) [reply files/zh-cn/web/api/abortsignal/index.md]: No changes?

File: files/zh-cn/web/api/abortsignal/index.md [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+### 事件
+
+使用 [`addEventListener()`](/zh-CN/docs/Web/API/EventTarget/addEventListener) 或将事件监听器分配给该接口的 `oneventname` 属性。
+
+- [`abort`](/zh-CN/docs/Web/API/AbortSignal/abort_event)
+  - : 当与 signal 通信 DOM 请求被中止时调用。也可以通过 `onabort` 属性调用。
+
+## 示例
+
+### 使用显示的信号终止 fetch 操作
```
Comment (Jack-Zhang-1314) [inline files/zh-cn/web/api/abortsignal/index.md:41]:
```suggestion
### 使用显式的信号终止 fetch 操作
```

File: files/zh-cn/web/api/abortsignal/index.md [hunk 13]
```diff
@@ -0,0 +1,141 @@
+---
+title: AbortSignal
+slug: Web/API/AbortSignal
+---
+{{APIRef("DOM")}}
+
+**`AbortSignal`** 接口表示一个信号对象（signal object），它允许您通过 {{domxref("AbortController")}} 对象与 DOM 请求（如 Fetch）进行通信并在需要时将其中止。
```
Comment (Jack-Zhang-1314) [inline files/zh-cn/web/api/abortsignal/index.md:7]:
```suggestion
**`AbortSignal`** 接口表示一个信号对象（signal object），它允许你通过 {{domxref("AbortController")}} 对象与 DOM 请求（如 Fetch）进行通信并在需要时将其中止。
```

File: files/zh-cn/web/api/abortsignal/index.md [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+**`AbortSignal`** 接口表示一个信号对象（signal object），它允许你通过 {{domxref("AbortController")}} 对象与 DOM 请求（如 Fetch）进行通信并在需要时将其中止。
+
+## 属性
+
+_AbortSignal 接口还继承了其父接口 {{domxref("EventTarget")}} 的属性。_
+
+- {{domxref("AbortSignal.aborted")}} {{readonlyInline}}
+  - : 一个 {{domxref("Boolean")}}，表示与之通信的请求是否被终止（`true`）或未终止（`false`）。
+- {{domxref("AbortSignal.reason")}} {{readonlyInline}}
+  - : 一旦信号被中止，提供一个 JavaScript 值表示的中止原因。
```
Comment (Jack-Zhang-1314) [inline files/zh-cn/web/api/abortsignal/index.md:16]:
```suggestion
  - : 一旦信号被中止，提供一个使用 JavaScript 值表示的中止原因。
```

File: files/zh-cn/web/api/abortsignal/index.md [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+
+## 示例
+
+### 使用显式的信号终止 fetch 操作
+
+以下片段展示了我们如何使用 signal 去中止使用 [Fetch API](/zh-CN/docs/Web/API/Fetch_API) 下载视频。
+
+我们首先使用 {{domxref("AbortController.AbortController", "AbortController()")}} 创建一个中止控制器，然后使用 {{domxref("AbortController.signal")}} 属性获取与它关联的 {{domxref("AbortSignal")}} 对象的引用。
+
+当 [fetch 请求](/zh-CN/docs/Web/API/fetch)开始时，我们将 `AbortSignal` 作为一个选项传递进请求的 option 对象中（见下面的 `{signal}`）。这个将信号和控制器与 fetch 请求相关联，并且允许我们通过调用 {{domxref("AbortController.abort()")}} 中止它。你可以看见当中止按钮（`abortBtn`）被点击，第二个事件监听器触发，fetch 操作被中止。
```
Comment (Jack-Zhang-1314) [inline files/zh-cn/web/api/abortsignal/index.md:47]:
```suggestion
当 [fetch 请求](/zh-CN/docs/Web/API/fetch)开始时，我们将 `AbortSignal` 作为一个选项传递进请求的 option 对象中（见下面的 `{signal}`）。这个将信号和控制器与 fetch 请求相关联，并且允许我们通过调用 {{domxref("AbortController.abort()")}} 中止它。你可以看见当中止按钮（`abortBtn`）被点击时，第二个事件监听器触发，使 fetch 操作被中止。
```

File: files/zh-cn/web/api/abortsignal/index.md [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+  } else {
+    // A network error, or some other problem.
+    console.error(`Error: type: ${err.name}, message: ${err.message}`);
+  }
+}
+```
+
+### 超时或显式中止 fetch
+
+`fetch()` 不是为了组合多个信号而设计的，所以你不能因为调用了 {{domxref("AbortController.abort()")}} 或者 `AbortSignal` 超时而"直接"中止下载（尽管在上面的例子中，如果由停止按钮等内置浏览器机制触发，超时信号将中止）。
```
Comment (Jack-Zhang-1314) [inline files/zh-cn/web/api/abortsignal/index.md:104]:
```suggestion
`fetch()` 不是为了组合多个信号而设计的，所以你不能因为调用了 {{domxref("AbortController.abort()")}} 或者 `AbortSignal` 超时而"直接"中止下载（尽管在上面的例子中，超时信号中止由停止按钮等内置浏览器机制触发）。
```

**Review by yin1999 — APPROVED** [review]

**Pull request merged**